### PR TITLE
[scudo] Use MemMap in BufferPool and RegionPageMap

### PR DESCRIPTION
**Pull request by fabio-d**

### CODE REVIEW

**Review by ChiaHungDuan — COMMENTED** [review]
Can you separate this into 2 CLs? The first one fixes the use of static buffer as,
in RegionPageMap,
```
  static const uptr StaticBufferSize = 512U * sizeof(uptr);
  static BufferPool<StaticBufferCount, StaticBufferSize> Buffers;
```
and in the `BufferPool`, 
```
static constexpr uptr NumBufferElements = (StaticBufferSize + sizeof(uptr)) / sizeof(uptr);
uptr RawBuffer[StaticBufferCount * NumBufferElements] GUARDED_BY(Mutex);
```
I would like to keep the `BufferSize` as the number of bytes.

**Comment by ChiaHungDuan** [comment]
Per discussion, we will use the number of elements and assign the type of buffer

**Comment by fabio-d** [comment]
Thanks! I've split the CL in two parts and uploaded the first part, with the changes we discussed, as #66896

Let's put this CL on hold for now until we land the new one :)

File: compiler-rt/lib/scudo/standalone/release.h
Comment (ChiaHungDuan) [inline compiler-rt/lib/scudo/standalone/release.h:154]: DCHECK_NE(Buf.Data, nullptr)

File: compiler-rt/lib/scudo/standalone/release.h
Comment (ChiaHungDuan) [inline compiler-rt/lib/scudo/standalone/release.h:166]: DCHECK_NE(Buf.Data, nullptr)

**Review by ChiaHungDuan — APPROVED** [review]

File: compiler-rt/lib/scudo/standalone/release.h
Comment (fabio-d) [reply compiler-rt/lib/scudo/standalone/release.h]: Done, thanks!

**Pull request merged**
